apps/comps/components/create-agent/index.tsx

```diff
@@ -0,0 +1,202 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useNavigationGuard } from "next-navigation-guard";
+import Image from "next/image";
+import { useRouter } from "next/navigation";
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { z } from "zod";
+
+import { Form } from "@recallnet/ui2/components/form";
+
+import { useRedirectTo } from "@/hooks/useRedirectTo";
+import { Agent } from "@/types/agent";
+import { asOptionalStringWithoutEmpty } from "@/utils";
+
+import { UnsavedChangesModal } from "../modals/unsaved-changes-modal";
+import { AgentCard } from "../user-agents";
+import { AgentCreated } from "./agent-created";
+import { BasicsStep } from "./basics-step";
+import { SocialsStep } from "./socials-step";
+import { Steps } from "./steps";
+
+const formSchema = z
+  .object({
+    name: z.string().min(1, "Agent name is required"),
+    imageUrl: asOptionalStringWithoutEmpty(
+      z.string().url({ message: "Must be a valid URL" }),
+    ),
+    repositoryUrl: z.string().url({ message: "Must be a valid URL" }),
```

Comment:
let's make the `repositoryUrl` optional. (this change was buried in some comments.)